ocaml/runtime/startup_aux.c

```diff
@@ -48,11 +50,25 @@ static void init_startup_params(void)
   char_os * cds_file;
 #endif
 
+  uintnat init_main_stack_wsz;
+  struct rlimit rlimit;
+  if (getrlimit(RLIMIT_STACK, &rlimit)) {
+    init_main_stack_wsz = Wsize_bsize(Stack_init_bsize);
```

Comment:
I think `Stack_init_bsize` might be too small, we should discuss and choose a larger number I think.

Reply:
We should also check this new number against `Max_stack_def`.

Reply:
Added the logic to cap the value to `Max_stack_def`.